CMakeLists.txt

```diff
@@ -196,10 +197,15 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 SET (Kokkos_ENABLE_AGGRESSIVE_VECTORIZATION ON CACHE BOOL
      "Kokkos aggressive vectorization")
 
-# Tell Kokkos we need lambdas in Cuda.
+# Check that gpu devices are actually detected
+set(NUM_GPU_DEVICES_PER_NODE "1" CACHE STRING "Number of gpu devices to use when testing if built with Kokkos_ENABLE_CUDA")
```

Comment:
Might want to declare this publicly as an option, as we may end up wanting to run tests on multiple GPUs.

Reply:
So the option command is only really appropriate for bool values. Is this what you are suggesting. By setting a value with CACHE it is publicly accessible. And that variable is documented in the cmake config table for parthenon. Is this what you meant?